### PR TITLE
NMS-14144: Unique filename suffix

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://issues.opennms.org/browse/NMS-14144

### CODE REVIEW

File: features/device-config/retrieval/src/main/java/org/opennms/features/deviceconfig/retrieval/impl/RetrieverImpl.java [hunk 1]
```diff
@@ -221,9 +223,14 @@ static String timeoutFailureMsg(String host, int port) {
...
+    private static byte[] uuidToBytes(UUID uuid) {
+        ByteBuffer bb = ByteBuffer.wrap(new byte[16]);
+        bb.putLong(uuid.getMostSignificantBits());
+        bb.putLong(uuid.getLeastSignificantBits());
+        return bb.array();
     }
 
-    private static AtomicLong uploadCounter = new AtomicLong();
+    private static String uniqueFilenameSuffix() {
+        return BASE64_URL_ENCODER.encodeToString(uuidToBytes(UUID.randomUUID()));
```
Comment (cgorantla) [inline features/device-config/retrieval/src/main/java/org/opennms/features/deviceconfig/retrieval/impl/RetrieverImpl.java:234]: Isn't `UUID.ramdomUUID().toString()` sufficient for this ?

**Review by cgorantla — COMMENTED** [review]
We may need to keep `monitor` prefix in order not to consume from sink module.
Relavant code [here](https://github.com/OpenNMS/opennms/blob/8c64617dbafeeb084b381511eba693dd23a3c7e2/features/device-config/sink/dispatcher/src/main/java/org/opennms/features/deviceconfig/sink/dispatcher/DeviceConfigDispatcher.java#L71)

File: features/device-config/retrieval/src/main/java/org/opennms/features/deviceconfig/retrieval/impl/RetrieverImpl.java (hunk 1 above)
Comment (christianpape) [reply features/device-config/retrieval/src/main/java/org/opennms/features/deviceconfig/retrieval/impl/RetrieverImpl.java:234]: @fooker suggested it: it is just shorter, e.g. ZuNXhDqFTXG_MSkUxBaTOQ vs. 66e35784-3a85-4d71-bf31-2914c4169339

**Comment by christianpape** [comment]
> We may need to keep `monitor` prefix in order not to consume from sink module.
> Relavant code [here](https://github.com/OpenNMS/opennms/blob/8c64617dbafeeb084b381511eba693dd23a3c7e2/features/device-config/sink/dispatcher/src/main/java/org/opennms/features/deviceconfig/sink/dispatcher/DeviceConfigDispatcher.java#L71)

done.

File: features/device-config/retrieval/src/main/java/org/opennms/features/deviceconfig/retrieval/impl/RetrieverImpl.java [hunk 2]
```diff
@@ -221,9 +223,14 @@ static String timeoutFailureMsg(String host, int port) {
...
-        return "monitor" + String.valueOf(uploadCounter.incrementAndGet());
+    private static byte[] uuidToBytes(UUID uuid) {
+        ByteBuffer bb = ByteBuffer.wrap(new byte[16]);
+        bb.putLong(uuid.getMostSignificantBits());
+        bb.putLong(uuid.getLeastSignificantBits());
+        return bb.array();
     }
 
-    private static AtomicLong uploadCounter = new AtomicLong();
+    private static String uniqueFilenameSuffix() {
```
Comment (cgorantla) [inline features/device-config/retrieval/src/main/java/org/opennms/features/deviceconfig/retrieval/impl/RetrieverImpl.java:233]: Add a comment to make sure `monitor` will not be removed in future.

Comment (christianpape) [reply features/device-config/retrieval/src/main/java/org/opennms/features/deviceconfig/retrieval/impl/RetrieverImpl.java:233]: done.

**Review by cgorantla — APPROVED** [review]

**Pull request merged**